qiskit/validation/base.py

```diff
@@ -151,6 +151,8 @@ def dump_additional_data(self, valid_data, original_data, **kwargs):
         else:
             additional_keys = set(original_data.__dict__) - set(valid_data)
             for key in additional_keys:
+                if key.startswith('_'):
+                    continue
```

Comment:
Can you also apply a similar tweak to the other for in line 149? Mostly for making the behavior a bit more consistent  - the other code path can also be triggered under certain conditions (ie. if the field is defined as being able to contain many).

Reply:
done